src/main/java/TaskList.java

```diff
@@ -0,0 +1,290 @@
+                } else {
+                    //"OOPS!!! Deadline in wrong format. Please use: deadline your_deadline /by YYYY-MM-DD
+                    // TTTT"
+                    throw new GrapieExceptions(ErrorMsg.deadlineFormatError);
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+            }
+        } else {
+            throw new GrapieExceptions(ErrorMsg.wakarimasenError);
+        }
+
+    }
+
+    /**
+     * Check if a valid number is inputted in String form.
+     *
+     * @param numStr the string to be checked.
+     * @return Boolean stating if the String is a valid number.
+     */
+    public boolean isNumber(String numStr) {
+        try {
+            Integer.parseInt(numStr);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Help mark a task as completed (O).
+     * Also checks if its a valid task number or if the task is already completed.
+     *
+     * @param doneTaskStr Task the user wants to be marked as complete.
+     * @throws GrapieExceptions Invalid task number, or already completed task thrown as error.
+     * @throws IOException      Throws away the exception.
+     */
+    public String completeTask(String doneTaskStr) throws GrapieExceptions, IOException {
+        if (doneTaskStr.length() <= 5) {
+            //no number behind
+            throw new GrapieExceptions(ErrorMsg.invalidNumberError);
+        } else {
+            //remember to add check for already completed task
+            String strNumberDone = doneTaskStr.substring(5, doneTaskStr.length());
+            strNumberDone.replaceAll("\\s+", ""); //remove all white spaces
+
+            boolean isANumber = isNumber(strNumberDone);
+
+            if (isANumber) {
+                int numDone = Integer.parseInt(strNumberDone); //convert to number
+                if (storingList.size() >= numDone && numDone != 0) {
+                    if (storingList.get(numDone - 1).isDone) {
+                        //if already true
+                        throw new GrapieExceptions(ErrorMsg.taskNumIsAlreadyDone(numDone));
+                    } else {
+
+                        storingList.get(numDone - 1).isDone = true;
+                        //storingList.set(taskNum - 1, updatedTask);
+
+                        String printStr =
+                                "Nice! I've marked this task as done: \n" + storingList.get(numDone - 1);
+                        //formattingDivider(printStr);
+                        storage.replaceLineFromHardDisk(numDone);
+
+                        return printStr;
+
+                    }
+                } else {
+                    throw new GrapieExceptions(ErrorMsg.numberDoNotExistError(numDone));
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.invalidNumberError);
+            }
+        }
```

Comment:
This seems more like your main logic. You might want to consider having your input parsed into an int beforehand, so that your TaskList can focus on ordinary List operations.